frontend/src/metabase/core/components/Sortable/SortableList.tsx

```diff
@@ -0,0 +1,126 @@
+import type React from "react";
+import { useState, useMemo, useEffect } from "react";
+import _ from "underscore";
+import type {
+  DragOverEvent,
+  DragStartEvent,
+  Modifier,
+  SensorDescriptor,
+} from "@dnd-kit/core";
+import { DndContext, DragOverlay } from "@dnd-kit/core";
+import { SortableContext, arrayMove } from "@dnd-kit/sortable";
+import { isNotNull } from "metabase/lib/types";
+import { Sortable, type SortableProps } from "./Sortable";
+
+type itemId = number | string;
+
+interface useSortableListProps<T> {
+  items: T[];
+  getId: (item: T) => itemId;
+  renderItem: (item: T, id?: itemId) => JSX.Element;
+  onSortStart?: (event: DragStartEvent) => void;
+  onSortEnd?: ({ id, newIndex }: { id: itemId; newIndex: number }) => void;
+  disableSort?: boolean;
+  sensors?: SensorDescriptor<any>[];
+  modifiers?: Modifier[];
+  sortableWrapper?: React.ComponentType<SortableProps>;
+  wrapperProps?: Record<any, any>;
+}
+
+export const SortableList = <T,>({
+  items,
+  getId,
+  renderItem,
+  onSortStart,
+  onSortEnd,
+  disableSort = false,
+  sensors = [],
+  modifiers = [],
+  sortableWrapper: SortableWrapper = Sortable,
+  wrapperProps,
```

Comment:
This feels a little bit off, what if we had to add `Sortable` as a part of `renderItem`? 

```tsx
import { SortableList } from "metabase/core/components/SortableList"

<SortableList
  items={items}
  renderItem={item => (
    <SortableList.Item key={item.id} id={item.id} draggingStyle={{ opacity: 0.5 }}>
      <MoreJSX />
    </SortableList.Item>
  )}
/>
```

// `SortableList.Item` is `Sortable` here